<a href="https://colab.research.google.com/github/francuzovd/probability_theory/blob/master/hw_lesson_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

1. Вероятность того, что стрелок попадет в мишень, выстрелив один раз, равна 0.8. Стрелок выстрелил 100 раз. Найдите вероятность того, что стрелок попадет в цель ровно 85 раз.

$P_n(X = k) = C_n^k*p^k*q^{n-k}$

In [1]:
import numpy as np

n = 100
k = 85
p = 0.8
q = 1 - p

P = round(np.math.factorial(n) / (np.math.factorial(k) * np.math.factorial(n - k)) * (p ** k) * (q ** (n - k)), 4)
P

0.0481

*Ответ :* 0.0481

2. Вероятность того, что лампочка перегорит в течение первого дня эксплуатации, равна 0.0004. В жилом комплексе после ремонта в один день включили 5000 новых лампочек. Какова вероятность, что ни одна из них не перегорит в первый день? Какова вероятность, что перегорят ровно две?

$P_m = {\lambda ^ m\over m!} * e^{-\lambda}$

In [2]:
n = 5000
p = 0.0004
lam = n * p
m = 0

P = round((lam ** m / np.math.factorial(m)) * np.exp(-lam), 4)
P

0.1353

In [3]:
n = 5000
p = 0.0004
lam = n * p
m = 2

P = round((lam ** m / np.math.factorial(m)) * np.exp(-lam), 4)
P

0.2707

*Ответ :* 

2.1 Вероятность, что не перегорит ни одна лампочка - 0.1353

2.2 Вероятность, что перегорит две лампочки - 0.2707

3. Монету подбросили 144 раза. Какова вероятность, что орел выпадет ровно 70 раз?

In [4]:
n = 144
k = 70
p = 0.5
q = 1 - p

P = round(np.math.factorial(n) / (np.math.factorial(k) * np.math.factorial(n - k)) * (p ** k) * (q ** (n - k)), 4)
P

0.0628

*Ответ: * 0.0628

4. В первом ящике находится 10 мячей, из которых 7 - белые. Во втором ящике - 11 мячей, из которых 9 белых. Из каждого ящика вытаскивают случайным образом по два мяча. Какова вероятность того, что все мячи белые? Какова вероятность того, что ровно два мяча белые? Какова вероятность того, что хотя бы один мяч белый?


Разберём вероятность, что все мяча белые. Для этог опосчитаем вероятность, что из каждого ящика достаём 2 белых мяча. После умножим вероятности.

In [5]:
a1 = np.math.factorial(7) / (np.math.factorial(2) * np.math.factorial(7 - 2))
b1 = np.math.factorial(10) / (np.math.factorial(2) * np.math.factorial(10 - 2))
P_box1 = a1 / b1
P_box1

0.4666666666666667

In [6]:
a2 = np.math.factorial(9) / (np.math.factorial(2) * np.math.factorial(9 - 2))
b2 = np.math.factorial(11) / (np.math.factorial(2) * np.math.factorial(11 - 2))
P_box2 = a2 / b2
P_box2

0.6545454545454545

In [7]:
P_com = P_box1 * P_box2
P_com

0.3054545454545455

*Ответ: * 0.305

Рассмотри вероятность, что вытащили только 2 белых мяча. Возможно несколькоо событий:

1.   Из первого ящика достали 2 белых мяча, из второго - 0.
2.   Из первого ящика достали 1 белый мяч, из второго - 1.
3.   Из первого ящика достали 0 белых мячей, из второго - 2.


Рассчитаем вероятность - Из первого ящика достали 2 белых мяча, из второго - 0.

In [8]:
a1 = np.math.factorial(7) / (np.math.factorial(2) * np.math.factorial(7 - 2))
b1 = np.math.factorial(10) / (np.math.factorial(2) * np.math.factorial(10 - 2))
P_box1 = a1 / b1
P_box1

0.4666666666666667

In [14]:
a2 = np.math.factorial(9) / (np.math.factorial(0) * np.math.factorial(9 - 0))
b2 = np.math.factorial(11) / (np.math.factorial(2) * np.math.factorial(11 - 2))
P_box2 = a2 / b2
P_box2

0.01818181818181818

In [15]:
P_com1 = P_box1 * P_box2
P_com1

0.0028282828282828283

Рассчитаем вероятность - Из первого ящика достали 1 белый мяч, из второго - 1.

In [13]:
a1 = np.math.factorial(7) / (np.math.factorial(1) * np.math.factorial(7 - 1))
b1 = np.math.factorial(10) / (np.math.factorial(2) * np.math.factorial(10 - 2))
P_box1 = a1 / b1
P_box1

0.15555555555555556

In [16]:
a2 = np.math.factorial(9) / (np.math.factorial(1) * np.math.factorial(9 - 1))
b2 = np.math.factorial(11) / (np.math.factorial(2) * np.math.factorial(11 - 2))
P_box2 = a2 / b2
P_box2

0.16363636363636364

In [17]:
P_com2 = P_box1 * P_box2
P_com2

0.025454545454545455

Рассчитаем вероятность - Из первого ящика достали 0 белых мячей, из второго - 2.

In [18]:
a1 = np.math.factorial(7) / (np.math.factorial(0) * np.math.factorial(7 - 0))
b1 = np.math.factorial(10) / (np.math.factorial(2) * np.math.factorial(10 - 2))
P_box1 = a1 / b1
P_box1

0.022222222222222223

In [19]:
a2 = np.math.factorial(9) / (np.math.factorial(2) * np.math.factorial(9 - 2))
b2 = np.math.factorial(11) / (np.math.factorial(2) * np.math.factorial(11 - 2))
P_box2 = a2 / b2
P_box2

0.6545454545454545

In [20]:
P_com3 = P_box1 * P_box2
P_com3

0.014545454545454545

Сложим вероятности  и получим вероятность вытащить 2 шара из 2х коробок

In [22]:
P_com = round(P_com1 + P_com2 + P_com3, 3)
P_com

0.043

*Ответ: * 0.043

Чтобы рассчитать вероятность, что хотя бы один мяч белы, мы рассчитаем вероятность, что вытянили 2 небелых мяча, а после вычтем эту вероятность из 1.

In [29]:
a1 = np.math.factorial(3) / (np.math.factorial(2) * np.math.factorial(3 - 2))
b1 = np.math.factorial(10) / (np.math.factorial(2) * np.math.factorial(10 - 2))
P_box1 = a1 / b1
P_box1

0.06666666666666667

In [30]:
a2 = np.math.factorial(2) / (np.math.factorial(2) * np.math.factorial(2 - 2))
b2 = np.math.factorial(11) / (np.math.factorial(2) * np.math.factorial(11 - 2))
P_box2 = a2 / b2
P_box2

0.01818181818181818

In [31]:
P_com = round(1 - P_box1 * P_box2, 4)
P_com

0.9988

*Ответ: * 0.9988